scylla/src/transport/session.rs

```diff
@@ -1056,10 +1056,19 @@ impl Session {
     }
 
     pub async fn check_schema_agreement(&self) -> Result<bool, QueryError> {
-        self.schema_agreement_auxilary(|connection: Arc<Connection>| async move {
-            connection.check_schema_agreement().await
-        })
-        .await
+        let connections = self.cluster.get_working_connections().await?;
+
+        let handles = connections.iter().map(|c| c.fetch_schema_version());
+        let mut results = join_all(handles).await;
+        let local_version: Uuid = self.fetch_schema_version().await?;
+
+        while let Some(res) = results.pop() {
+            let version: Uuid = res?;
+            if version != local_version {
+                return Ok(false);
+            }
+        }
+        Ok(true)
```

Comment:
It looks like this loop could be replaced with some more idiomatic one-liner, but I'm also fine with it as is; @piodul please ack/

Reply:
Done